src/oldtime.rs

```diff
@@ -229,6 +229,12 @@ impl Duration {
         if d < MIN || d > MAX { None } else { Some(d) }
     }
 
+    /// Returns the duration as an absolute (non-negative) value.
+    #[inline]
+    pub fn abs(&self) -> Duration {
+        Duration { secs: self.secs.abs(), nanos: self.nanos }
```

Comment:
This seems like it doesn't handle nanos properly; -1s + .5s is -0.5s which should have an absolute value of .5s not 1.5s

Reply:
You're correct: I didn't realize at the time that the `nanos` field always 'adds' to the `secs` field (which makes sense since you can't store `-0` in the `secs` field).

Reply:
I submitted #734.